lib/mail_room/mailbox_handler.rb

```diff
@@ -45,17 +43,19 @@ def new_messages
     # search for all new (unseen) message ids
     # @return [Array<Integer>] message ids
     def new_message_ids
-      @imap.search(@mailbox.search_command)
+      uids = @imap.uid_search(@mailbox.search_command)
```

Comment:
Could we do this without the local variable? I like that functional style :smile: 

`@imap.uid_search(@mailbox.search_command).select { |uid| @mailbox.delivery?(uid) }`